modules/sor/lib/poolsV3/reClamm/reClammPool.ts

```diff
@@ -0,0 +1,176 @@
+import { Address, Hex, parseEther, parseUnits } from 'viem';
+import { Token } from '@balancer/sdk';
+import { HookState, ReClammState } from '@balancer-labs/balancer-maths';
+import { Chain } from '@prisma/client';
+
+import { PrismaPoolAndHookWithDynamic } from '../../../../../prisma/prisma-types';
+import { chainToChainId as chainToIdMap } from '../../../../network/chain-id-to-chain';
+import { TokenPairData } from '../../../../sources/contracts/v3/fetch-tokenpair-data';
+
+import { WAD } from '../../utils/math';
+import { BasePoolMethodsV3 } from '../basePoolMethodsV3';
+import { PoolTokenWithRate } from '../../utils/poolTokenWithRate';
+import { Erc4626PoolToken } from '../../utils/erc4626PoolToken';
+
+import { getHookState } from '../../utils/helpers';
+
+import { LiquidityManagement } from '../../../types';
+import { BasePoolV3 } from '../basePoolV3';
+import { ReclammData } from '../../../../pool/subgraph-mapper';
+import { ReClammParams } from './types';
+
+type ReClammPoolToken = PoolTokenWithRate | Erc4626PoolToken;
+
+export class ReClammPool extends BasePoolV3 implements BasePoolMethodsV3 {
+    public readonly poolType = 'RECLAMM';
+    public readonly reClammParams: ReClammParams;
+
+    public tokens: ReClammPoolToken[];
+
+    private readonly tokenMap: Map<string, ReClammPoolToken>;
+
+    static fromPrismaPool(
+        pool: PrismaPoolAndHookWithDynamic,
+        underlyingTokens: { address: string; decimals: number }[] = [],
+        currentTimestamp: bigint,
+    ): ReClammPool {
+        const poolTokens: ReClammPoolToken[] = [];
+
+        if (!pool.dynamicData) throw new Error('Stable pool has no dynamic data');
```

Comment:
Should be ReClamm pool. (theres a few more like this)

Reply:
Done! Adapted it so reusing existing pool implementation as base for new pool types won't cause issues like that in the future.